speaker/main.go

```diff
@@ -69,17 +70,19 @@ func main() {
 	}
 
 	var (
-		config      = flag.String("config", "config", "Kubernetes ConfigMap containing MetalLB's configuration")
+		configMap   = flag.String("config", "config", "Kubernetes ConfigMap containing MetalLB's configuration")
 		configNS    = flag.String("config-ns", "", "config file namespace (only needed when running outside of k8s)")
 		kubeconfig  = flag.String("kubeconfig", "", "absolute path to the kubeconfig file (only needed when running outside of k8s)")
-		host        = flag.String("host", os.Getenv("METALLB_HOST"), "HTTP host address")
+		metricsHost = flag.String("metrics-host", os.Getenv("METALLB_METRICS_HOST"), "Bind address for Prometheus metrics")
+		metricsPort = flag.Int("metrics-port", 80, "Bind port for Prometheus metrics")
 		mlBindAddr  = flag.String("ml-bindaddr", os.Getenv("METALLB_ML_BIND_ADDR"), "Bind addr for MemberList (fast dead node detection)")
 		mlBindPort  = flag.String("ml-bindport", os.Getenv("METALLB_ML_BIND_PORT"), "Bind port for MemberList (fast dead node detection)")
 		mlLabels    = flag.String("ml-labels", os.Getenv("METALLB_ML_LABELS"), "Labels to match the speakers (for MemberList / fast dead node detection)")
 		mlNamespace = flag.String("ml-namespace", os.Getenv("METALLB_ML_NAMESPACE"), "Namespace of the speakers (for MemberList / fast dead node detection)")
 		mlSecret    = flag.String("ml-secret-key", os.Getenv("METALLB_ML_SECRET_KEY"), "Secret key for MemberList (fast dead node detection)")
 		myNode      = flag.String("node-name", os.Getenv("METALLB_NODE_NAME"), "name of this Kubernetes node (spec.nodeName)")
-		port        = flag.Int("port", 80, "HTTP listening port")
+		statusHost  = flag.String("status-host", os.Getenv("METALLB_STATUS_HOST"), "Bind address for status endpoint")
```

Comment:
The default behavior right now is that if you don't specify a hostname, `:7473` is used as the addressed passed to `http.ListenAndServe()`, which results in the speaker listening on all of the **pod's** interfaces. In practice, this simply means that by default a speaker pod is exposing the status endpoint, however you still need to use `kubectl port-forward` or some other mechanism to get network connectivity to the pod, i.e. the status endpoint shouldn't be available from the public internet.

At the **code** level it could make sense to me to set the default to `127.0.0.1`, however:

1. I've copied the behavior from the metrics endpoint so we would have to change that, too.
1. If we set the default to `127.0.0.1` at the code level, we should probably override the default in the deployment manifest since the status endpoint is unusable when it's listening on localhost.

Reply:
OK, I think this is fine as-is

Reply:
> which results in the speaker listening on all of the **pod's** interfaces

The pod [uses hostNetwork][link], though. I guess it should be fine, I don't see a reason either to treat the metrics endpoint and this endpoint differently (unless BGP password is exposed or something else) 

[link]: https://github.com/metallb/metallb/blob/main/manifests/metallb.yaml#L336